cpp/src/components/connectivity.cu

```diff
@@ -41,190 +40,71 @@ namespace detail {
-               <<"; p_d_col_ind valid: " << (p_d_col_ind != nullptr)
-               <<"\n";
-      
-      std::cout<<"############## d_ro:\n";
-      print_v(d_ro, std::cout);
-
-      std::cout<<"############## d_ci:\n";
-      print_v(d_ci, std::cout);
-#endif
-
+  if (connectivity_type == cugraph_cc_t::CUGRAPH_WEAK) {
       //check if graph is undirected; return w/ error, if not?
       //Yes, for now; in the future we may remove this constraint; 
       //
-      bool is_symmetric = cugraph::detail::check_symmetry(nrows, p_d_row_offsets, nnz, p_d_col_ind);
-#ifdef _DEBUG_WEAK_CC
-      std::cout<<"############## "
-               <<"; adj. matrix symmetric? " << is_symmetric
-               <<"\n";
-#endif
+      bool is_symmetric = cugraph::detail::check_symmetry(graph.number_of_vertices, graph.offsets, graph.number_of_edges, graph.indices);
+
+      auto d_alloc = std::shared_ptr<MLCommon::deviceAllocator>{new MLCommon::defaultDeviceAllocator()};
       
       CUGRAPH_EXPECTS( is_symmetric, "Invalid API parameter: graph must be symmetric");
-      MLCommon::Sparse::weak_cc_entry<IndexT, TPB_X>(p_d_labels,
-                                                     p_d_row_offsets,
-                                                     p_d_col_ind,
-                                                     nnz,
-                                                     nrows,
+      MLCommon::Sparse::weak_cc_entry<VT, ET, TPB_X>(labels,
+                                                     graph.offsets,
+                                                     graph.indices,
+                                                     graph.number_of_edges,
+                                                     graph.number_of_vertices,
+                                                     d_alloc,
                                                      stream);
-
-    }
-  else
-    {
-      
-      //device memory requirements: 2n^2 + 2n x sizeof(IndexT) + 1 (for flag)
-      //( n = |V|)
-      //
-      size_t n2 = 2*nrows;
-      n2 = n2*(nrows*sizeof(ByteT) + sizeof(IndexT)) + 1;
-
-      int device;
-      cudaDeviceProp prop;
+  } else {
+    //device memory requirements: 2n^2 + 2n x sizeof(IndexT) + 1 (for flag)
+    //( n = |V|)
+    //
+    size_t n2 = 2 * graph.number_of_vertices;
+    n2 = n2 * (graph.number_of_vertices * sizeof(ByteT) + sizeof(VT)) + 1;
+
+    int device;
+    cudaDeviceProp prop;
       
-      cudaGetDevice(&device);
-      cudaGetDeviceProperties(&prop, device);
+    cudaGetDevice(&device);
+    cudaGetDeviceProperties(&prop, device);
 
-      if( n2 > prop.totalGlobalMem )
-        {
-
-          CUGRAPH_FAIL("ERROR: Insufficient device memory for SCC");
-        }
-      SCC_Data<ByteT, IndexT> sccd(nrows, p_d_row_offsets, p_d_col_ind);
-      sccd.run_scc(p_d_labels);
-      
+    if( n2 > prop.totalGlobalMem ) {
+      CUGRAPH_FAIL("ERROR: Insufficient device memory for SCC");
```

Comment:
And this is a bug, this just checks the total physical DRAM size is sufficient to run SCC (assuming that no memory is allocated for anything else). We have no reason to check this just for SCC, better just throw std::bad_alloc if memory allocation fails (assuming that this code is written in a exception safe way... this is a big if and our SCC algorithm requires impractically large memory for large graphs, we need to redo the SCC code anyways, so I am against spending too much time on the current SCC code, but just in case there is a quick fix).

Reply:
I haven't really looked at the SCC implementation.  I'd be fine with removing this check and just let it fail if there is insufficient memory.

Reply:
Either works for me, our SCC implementation has issues and I think we need to redo SCC anyways... Better not spend too much effort on this.